### PR TITLE
fixed predictions refresh for favorites page (and various other related cleanup)

### PR DESCRIPTION
**Pull request by tinzerillo**

### CODE REVIEW

File: static/controllers/favoritesController.js [hunk 1]
```diff
@@ -1,94 +1,94 @@
-app.controller('favoritesController', function ($scope, $localStorage, $http, FavoritesService, UtilityService) {
-    'use strict';
+app.controller('favoritesController', function ($scope, $localStorage, $http, FavoritesService, UtilityService, $interval, PredictionService) {
+    var vm = this;
```
Comment (mathcolo) [inline static/controllers/favoritesController.js:2]: Just curious if this is actually preferable to using `this` everywhere? heh

File: static/controllers/favoritesController.js [hunk 2]
```diff
@@ -1,94 +1,94 @@
...
+
+	var refreshFavoritesData = function() {
+	    for (var i = 0; i < vm.stations.length; i++) {
+            refreshOutboundPredictions(i);
+            refreshInboundPredictions(i);
+		}
+	};
+
+	var refreshInboundPredictions = function(station_idx) {
+	    $http.get('/station/' + vm.stations[station_idx].id + '/direction/' + 1 + '/nextservice')
```
Comment (mathcolo) [inline static/controllers/favoritesController.js:47]: I have continued concerns over the number of http requests this involves but I'll be "that guy" in another ticket 👍  👻

File: static/controllers/favoritesController.js [hunk 3]
```diff
@@ -1,94 +1,94 @@
...
 
-			});
-			}
-		}
+		});
 	}
 	
-	$scope.init();
-	
+	init();
```
Comment (mathcolo) [inline static/controllers/favoritesController.js:85]: [Don't actually care] Should the init function and call be at the top just so if someone new ever opened the file, they'd see that first?

File: static/app.js [hunk 4]
```diff
@@ -16,20 +16,28 @@ app.controller('appCtrl', function ($http, $mdSidenav, $mdDialog, $location) {
...
+
 
 // configure our routes
 app.config(function($routeProvider) {
 	$routeProvider
 
 		.when('/trains/:route_name', {
 			templateUrl : 'static/partials/route.html',
-			controller  : 'routeController'
+			controller: '',
```
Comment (mathcolo) [inline static/app.js]: Ah, angular readability! If this had to happen, I am so sorry

File: static/controllers/favoritesController.js (hunk 3 above)
Comment (tinzerillo) [inline static/controllers/favoritesController.js:85]: Yes I think that would be good!

File: static/controllers/favoritesController.js (hunk 2 above)
Comment (tinzerillo) [inline static/controllers/favoritesController.js:47]: Yeah we can make a ticket for that! I figured it was outside the scope of this particular issue.

File: static/controllers/favoritesController.js (hunk 1 above)
Comment (tinzerillo) [inline static/controllers/favoritesController.js:2]: Actually necessary surprisingly! To keep the scope consistent (https://johnpapa.net/angularjss-controller-as-and-the-vm-variable/)

File: static/app.js (hunk 4 above)
Comment (tinzerillo) [inline static/app.js]: Because I had the controller defined in the html it was instantiating it twice >.<

Comment (mathcolo) [inline static/app.js]: Gaah OK. Can this key be eliminated or is it required?

Or eliminated from the html? Just a bizarre thing :'(

Comment (tinzerillo) [inline static/app.js]: We need it in the html but I can try removing it here. I can't remember now if it would work! 100% hear you here though haha

**Comment by mathcolo** [comment]
Comments aside, +1! Nice job!

(Actually make that +2; there was more than 1 commit in the PR 👍  :'))

File: static/controllers/favoritesController.js (hunk 3 above)
Comment (tinzerillo) [inline static/controllers/favoritesController.js:85]: JK...due to the way javascript works...things have to be declared before you use them. leaving as-is

File: static/app.js (hunk 4 above)
Comment (tinzerillo) [inline static/app.js]: This is fixed!

**Pull request merged**